### PR TITLE
feat: tei relationships mvp

### PR DESCRIPTION
**Pull request by amcgee**
This is a barebones implementation of TEI Relationships in maps app.  It is missing a few nice-to-have features but should be functional as-is.

This requires https://github.com/dhis2/gis-api/pull/53 before it is merged, so I'm leaving this as a draft for now.

The features which have and have not been implemented here are:
- [x] Allow selection of a TEI Relationship Type in the TEI layer dialog
- [x] Allow styling of the relationship line and related entity point (or polygon) in the style tab
- [x] Display entity, relationship line, and related entity in the legend for a given TEI layer
- [x] Render relationship lines from primary TEI entities to secondary related TEI entities on the map
- [x] Support multi-level relationships (`parent->child/parent->child`)
- [x] Correctly process relationship loops
- [x] On-click popup for related entities
- [x] Render relationship lines from the centroid of polygons
    - [ ] Use the center-of-mass or the closest vertex instead
- [ ] Render arrows on relationship lines
- [ ] Don't overlap entity points and relationship lines (currently the lines begin at the center of the circle and are visible when the layer opacity is < 1)
- [ ] On-click highlighting of "index" entity and its relationships
- [ ] Relationship count in on-click popup (this shouldn't be difficult to add if we only include 1st-degree relationships)
- [ ] Remove on-hover styling from relationship lines
- [ ] Make relationship line weight configurable
- [ ] Filter the "secondary" TEI dataset so that un-related features are not displayed.
- [ ] Warn the user or forbid them from continuing when selecting a relationship type that will result in a very large dataset download.  Currently the entire source and target datasets are downloaded.
- [x] Add "experimental" feature warning to Relationships tab

TODO
- [ ] Ensure TEI layers can be saved and loaded successfully

### CODE REVIEW

**Comment by turban** [comment]
Can the TEI layers with relationships be saved? Maybe add to list above.

**Comment by turban** [comment]
It would be nice to be able to change line width.

File: src/components/edit/trackedEntity/TrackedEntityRelationshipTypeSelect.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        }));
+});
+
+class TrackedEntityRelationshipTypeSelect extends Component {
+    state = {
+        allTypes: null,
+        error: null,
+    };
+
+    componentDidMount() {
```
[the remark this answers is not in the document]
Comment (amcgee) [reply src/components/edit/trackedEntity/TrackedEntityRelationshipTypeSelect.js]: We could yes, we do a lot of mixing of redux epics and explicit API requests for data fetching.  I'd like to consolidate, but agree this is probably ok for now.

**Comment by amcgee** [comment]
This is the "experimental feature" warning I added to the Relationships tab of the TEI layer dialog:

Screenshot 2019-04-09 19 15 48

**Comment by amcgee** [comment]
Merging!

**Pull request merged**